kolibri/plugins/coach/assets/src/modules/examCreation/handlers.js

```diff
@@ -118,6 +124,56 @@ export function showExamCreationTopicPage(store, params) {
     });
   });
 }
+export function showExamCreationBookamrksPage(store, params) {
+  return store.dispatch('loading').then(() => {
+    const { topicId } = params;
+    const topicNodePromise = ContentNodeResource.fetchModel({ id: topicId });
+    const childNodesPromise = ContentNodeResource.fetchCollection({
+      getParams: {
+        parent: topicId,
+        kind_in: [ContentNodeKinds.TOPIC, ContentNodeKinds.VIDEO, ContentNodeKinds.EXERCISE],
+      },
+    });
+    const loadRequirements = [topicNodePromise, childNodesPromise];
+
+    return Promise.all(loadRequirements).then(([topicNode, childNodes]) => {
+      return filterAndAnnotateContentList(childNodes).then(() => {
+        store.commit('SET_TOOLBAR_ROUTE', {
+          name: PageNames.EXAMS,
+        });
+        return showExamCreationPage(store, {
+          classId: params.classId,
+          bookmarksList: childNodes,
+          pageName: PageNames.EXAM_CREATION_BOOKMARKS,
+          ancestors: [...topicNode.ancestors, topicNode],
+        });
+      });
+    });
+  });
+}
+export function showExamCreationAllBookmarks(store) {
+  return store.dispatch('loading').then(() => {
+    let dataArr = [];
+    getBookmarks()
+      .then(bookmarks => {
+        bookmarks.forEach(bookmark => {
+          ContentNodeResource.fetchModel({ id: bookmark.contentnode_id }).then(data => {
+            dataArr.push(data);
+          });
+        });
+      })
```

Comment:
Similarly to a previous comment, it seems like we should consider making an API endpoint (possibly in a follow-up PR) on the Bookmarks API that returns a combination of Bookmarks and their associated ContentNodes.

In the absence of that optimization, this could avoid making an API request for every bookmark using `fetchCollection`  with `id__in` like:

```
const contentNodeIds = bookmarks.map(b => b.contentnode_id);
ContentNodeResource.fetchCollection({id__in: contentNodeIds})
    .then(/* and so on */)
```

Which should get you all of the ContentNode data with one request.